controllers/bsl.go

```diff
@@ -416,16 +422,23 @@ func (r *DPAReconciler) ensureSecretDataExists(dpa *oadpv1alpha1.DataProtectionA
 				return fmt.Errorf("Secret name specified in BackupLocation %s cannot be empty", bsl.Name)
 			}
 		}
+
 		// Check if the BSL secret key configured in the DPA exists with a secret data
-		secretName, secretKey := r.getSecretNameAndKeyforBackupLocation(*bsl)
-		bslSecret, err := r.getProviderSecret(secretName)
-		if err != nil {
-			return err
+
+		if bsl.CloudStorage != nil {
```

Comment:
I am very confused here: why CloudStorage does not have the same checks for credentials as above? (checking if name and key are not empty)

Reply:
It should not be something like this?
```go
		// Check if the user specified credential under cloud storage
		if bsl.CloudStorage != nil && bsl.CloudStorage.Credential != nil {
			// Check if user specified empty credential key
			if bsl.CloudStorage.Credential.Key == "" {
				return fmt.Errorf("Secret key specified in CloudStorage of BackupLocation %s cannot be empty", bsl.Name)
			}
			// Check if user specified empty credential name
			if bsl.CloudStorage.Credential.Name == "" {
				return fmt.Errorf("Secret name specified in CloudStorage of BackupLocation %s cannot be empty", bsl.Name)
			}
			err := r.verifySecretContent(bsl.CloudStorage.Credential.Name, bsl.CloudStorage.Credential.Key)
			if err != nil {
				return err
			}
		}
```

Reply:
These Credential checks are also needed at other places, like in the ReconcileBackupStorageLocations , So i have included it in the below function at a common place "getSecretNameandKeyFromCloudStorage"

Reply:
But `getSecretNameandKeyFromCloudStorage` function https://github.com/openshift/oadp-operator/pull/1326/files?diff=split&w=0#diff-14341d832b55f374f353d693c68774d04617876f76c0de474e91763d14594614R246 

only runs `verifySecretContent` function https://github.com/openshift/oadp-operator/pull/1326/files?diff=split&w=0#diff-14341d832b55f374f353d693c68774d04617876f76c0de474e91763d14594614R786 

it does not validate non empty name and key

Reply:
Agree, Thank you!